WcaOnRails/spec/controllers/registrations_controller_spec.rb

```diff
@@ -161,6 +161,18 @@
       patch :update, id: registration.id, registration: { status: :accepted }
       expect(registration.reload.accepted?).to eq true
     end
+
+    it "can register for their own competition that is not yet visible" do
+      competition.update_column(:showAtAll, false)
+      expect(RegistrationsMailer).to receive(:notify_organizers_of_new_registration).and_call_original
```

Comment:
You can actually use `.with(...)` to specify what arguments you expect to be passed to `notify_organizers_of_new_registration`. See my comment to @jonatanklosko here: https://github.com/cubing/worldcubeassociation.org/pull/609/files/f238c37415fd95d9dec36ab72bfdb85052f5441a#r63247794